packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-credit-button-strategy.ts

```diff
@@ -98,6 +124,183 @@ export default class PaypalCommerceCreditButtonStrategy implements CheckoutButto
         }
     }
 
+    private async _onCancel() {
+        const state = this._store.getState();
+        const billingAddress = state.billingAddress.getBillingAddressOrThrow();
+        const resetAddress = this._resetAddress(billingAddress);
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(resetAddress));
+        await this._store.dispatch(this._consignmentActionCreator.updateAddress(resetAddress));
+        await this._updateOrder();
+    }
+
+    private _resetAddress(address: BillingAddressRequestBody) {
+        const { firstName, lastName, address1, email } = address;
+
+        return {
+            ...address,
+            firstName: firstName !== 'Fake' ? firstName : '',
+            lastName: lastName !== 'Fake' ? lastName : '',
+            address1: address1 !== 'Fake street' ? address1 : '',
+            email: email !== 'fake@fake.fake' ? email : '',
+        }
+    }
+
+    private async _onHostedCheckoutApprove(
+        data: ApproveCallbackPayload,
+        actions: ApproveCallbackActions,
+        methodId: string,
+    ): Promise<boolean> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const orderDetails = await actions.order.get();
+
+        if (cart.lineItems.physicalItems.length > 0) {
+            const address = this._getValidAddress(
+                orderDetails.payer.name,
+                orderDetails.payer.email_address,
+                orderDetails.purchase_units[0].shipping.address
+            );
+
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+            await this._store.dispatch(this._consignmentActionCreator.updateAddress(address));
+            await this._updateOrder();
+        } else {
+            const address = this._getValidAddress(
+                orderDetails.payer.name,
+                orderDetails.payer.email_address,
+                orderDetails.payer.address
+            );
+
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder({}, { params: { methodId } }));
+        await this._submitPayment(methodId, data.orderID);
+
+        return true;
```

Comment:
Do we need to return a boolean here? Or you can return the submitpayment promise and make decisions based on that?

Reply:
we have to return true because this is paypal callback and it waiting for true in order to trigger onComplete method after onApprove